symphony/lib/toolkit/class.widget.php

```diff
@@ -403,7 +403,7 @@ public static function TableHead(array $columns = null)
                     $th->setAttribute('scope', $col[1]);
                 }
 
-                if (is_array($col[2]) && !empty($col[2])) {
+                if (!empty($col[2]) && is_array($col[2])) {
```

Comment:
You make valid points ;)

> Is there any "official recommendation" for certain patterns?

I've yet to seen some especially for PHP. Maybe I should read more of the PSR-* stuff. I also watched a [Youtube video recently about legacy php apps](https://www.youtube.com/watch?v=jJ5PCbjZIsY) and the book seems interesting, but it is based on 3 books I've already read, which are not php related, but tells me I should create those methods, specially because they contains edge cases so they can be unit tested. But again, here some the naming problem.

.NET has a `String.isNullorEmpty` and `String.isNullOrWhitespace()` which is quite nice. Maybe `Array::isEmpty()` (or `General::isArrayNotEmpty()` would make it so that it's clear that it must be an array ?